cmake/dca_compiler_options.cmake

```diff
@@ -21,7 +21,7 @@
 set(DCA_WARNINGS -Wall -Wextra -Wpedantic -Wno-sign-compare -Wno-dangling-else)
 
 # Languange standard
-set(DCA_STD_FLAG -std=c++14)
+set(DCA_STD_FLAG -std=c++17)
```

Comment:
as in #160: not strictly necessary, but it would be nice.

Reply:
I think while you are here is a good time to make this transition.